Tasks/AzureIoTEdge/util.ts

```diff
@@ -0,0 +1,238 @@
+import Constants from "./constant";
+import * as tl from "azure-pipelines-task-lib/task";
+import * as crypto from "crypto";
+import { IExecSyncOptions } from 'azure-pipelines-task-lib/toolrunner';
+import { Writable } from "stream";
+import { RegistryCredential } from './registryCredentialFactory';
+
+interface Cmd {
+  path: string;
+  arg: string;
+  execOption: IExecSyncOptions;
+}
+
+export default class Util {
+  public static expandEnv(input: string, ...exceptKeys: string[]): string {
+    const pattern: RegExp = new RegExp(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g);
+    const exceptSet: Set<string> = new Set(exceptKeys);
+    return input.replace(pattern, (matched) => {
+      if (exceptKeys && exceptSet.has(matched)) {
+        return matched;
+      }
+      const key: string = matched.replace(/\$|{|}/g, "");
+      return process.env[key] || matched;
+    });
+  }
+
+  public static validateModuleJson(moduleJsonObject: any): void {
+    // Will throw error if parent property does not exist
+    if (moduleJsonObject.image.tag.platforms == undefined) {
+      throw new Error(`${Constants.fileNameModuleJson} image.tag.platforms not set`);
```

Comment:
should move these literals to task.json messages

Reply:
validateModuleJson and validateDeployTemplateJson method are no longer used and I'll remove them.
Moved the literal in `setTaskRootPath` method to task.json